### PR TITLE
Fix: Added Dropbox Personal previewRenderable check for record type

### PR DESCRIPTION
**Pull request by SharmaNishchay**
## Description

It implements the previewRenderable false for non file type records.

### Type of Change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

This PR has been tested manually by indexing, reindexing, moving records, updating the contents & querying the LLM.

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [X] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [X] **Search capabilities** are working correctly
- [X] **Knowledge search** is functioning properly
- [X] **Connector indexing** is working as expected
- [X] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [X] My code follows the project's coding standards
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [X] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [X] Authentication/authorization is properly handled
- [X] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

### CODE REVIEW

File: backend/python/app/connectors/sources/dropbox_individual/connector.py [hunk 1]
```diff
@@ -611,9 +612,11 @@ async def _process_dropbox_items_generator(
...
                 if record_update and record_update.record:
-                    if not self.indexing_filters.is_enabled(IndexingFilterKey.FILES, default=True):
-                        record_update.record.indexing_status = IndexingStatus.AUTO_INDEX_OFF.value
-                    if record_update.record.is_shared and not self.indexing_filters.is_enabled(IndexingFilterKey.SHARED, default=True):
+                    folder_disabled = not record_update.record.is_file
+                    files_disabled = not self.indexing_filters.is_enabled(IndexingFilterKey.FILES, default=True)
+                    shared_disabled = record_update.record.is_shared and not self.indexing_filters.is_enabled(IndexingFilterKey.SHARED, default=True)
+                    
+                    if folder_disabled or files_disabled or shared_disabled:
                         record_update.record.indexing_status = IndexingStatus.AUTO_INDEX_OFF.value
```
[the remark this answers is not in the document]
Comment (SharmaNishchay) [reply backend/python/app/connectors/sources/dropbox_individual/connector.py]: Moved the logic to ensure folders don't get indexed in the reindexing.

File: backend/python/app/connectors/sources/dropbox_individual/connector.py [hunk 2]
```diff
@@ -522,6 +522,7 @@ async def _process_dropbox_entry(
                 parent_external_record_id=parent_external_record_id,
                 size_in_bytes=entry.size if is_file else 0,
                 is_file=is_file,
+                preview_renderable=is_file,  # Only files should be previewable, not folders
```
[the remark this answers is not in the document]
Comment (SharmaNishchay) [reply backend/python/app/connectors/sources/dropbox_individual/connector.py]: Removed the sharing check in the indexing as sharing is not a feature of dropbox personal.
